src/Processors/Formats/Impl/NativeFormat.cpp

```diff
@@ -40,6 +40,12 @@ class NativeInputFormat final : public IInputFormat
         return Chunk(block.getColumns(), num_rows);
     }
 
+    void setReadBuffer(ReadBuffer & in_) override
+    {
+        reader.setReadBuffer(in_);
```

Comment:
@CurtizJ maybe it is easier to just recreate reader here, without other changes at all ?

Reply:
I agree, it's better.